includes/abstracts/abstract-wc-stripe-payment-gateway.php

```diff
@@ -1548,6 +1548,11 @@ public function confirm_intent( $intent, $order, $prepared_source ) {
 	 * @param stdClass $intent Payment intent information.
 	 */
 	public function save_intent_to_order( $order, $intent ) {
+		// Don't save any intent information on a subscription.
+		if ( $this->is_subscription( $order ) ) {
```

Comment:
I'm a little bit unsure about this change so I'd like to get others' opinions on it.
I can't ever see a reason we'd want to store an intent ID on the subscription because an intent is specific to the individual payments and/or parent order, not the subscription specifically.

Reply:
I tend to agree because as you mentioned subscription's don't typically store the set up intent. The only reason this happens is if you change a subscriptions payment method and the `process_payment()` function is passed a subscription rather than an order and so an setup intent gets saved to a subscription rather than an order. 

In that case, I don't think there's any reason the set up intent is stored on the subscription.